lib/ami_spec/aws_key_pair.rb

```diff
@@ -0,0 +1,36 @@
+require 'aws-sdk-ec2'
+require 'logger'
+require 'securerandom'
+require 'tempfile'
+require 'pathname'
+
+module AmiSpec
+  class AwsKeyPair
+
+    def self.create(**args)
+      new(**args).tap(&:create)
+    end
+
+    def initialize(ec2: Aws::EC2::Resource.new, key_name_prefix: 'ami-spec-', logger: Logger.new(STDOUT))
+      @ec2 = ec2
+      @key_name = "#{key_name_prefix}#{SecureRandom.uuid}"
+      @logger = logger
+    end
+
+    attr_reader :key_name, :key_file
+
+    def create
+      @logger.info "Creating temporary AWS key pair: #{@key_name}"
+      @key_pair = @ec2.create_key_pair(key_name: @key_name)
+      @temp_file = Tempfile.new('key')
```

Comment:
By using a `Tempfile` object, the private key will be deleted from the filesystem when the Ruby process terminates.